src/utils/utils.py

```diff
@@ -33,6 +33,51 @@
 _MOSES_DETOKENIZER = MosesDetokenizer()
 
 
+def check_for_previous_checkpoints(serialization_dir, tasks, phase, load_model):
+    task_directory = ""
+    for task in tasks[::-1]:
+        last_checkpoint = find_last_checkpoint_suffix(serialization_dir, phase, task.name)
+        # if we haven't found a task directory yet
+        if last_checkpoint is not None and len(task_directory) == 0 and phase == "target_train":
+            task_directory = task.name
+    checkpoint_pattern = os.path.join(serialization_dir, task_directory, "*_{}_*.th".format(phase))
+    if not load_model:
+        assert_for_log(
+            len(glob.glob(checkpoint_pattern)) == 0,
+            "There are existing checkpoints in %s which will be overwritten. "
+            "If you are restoring from a run, or would like to train from an "
+            "existing checkpoint, Use load_model = 1 to load the checkpoints instead. "
+            "If you don't want them, delete them or change your experiment name."
+            % serialization_dir,
+        )
+    return task_directory
+
+
+def find_last_checkpoint_suffix(serialization_dir, search_phase="pretrain", task_name=""):
```

Comment:
Alternately, we could put the test to check for a matching set of four checkpoints here, rather than above, and make sure this is called in target_train, too.